organizations/views.py

```diff
@@ -104,19 +113,29 @@ def get_object(self, queryset=None):
         Make sure that current user is allowed to delete the shift
         """
         shift = super(ShiftDeleteView, self).get_object()
-
         current_user = self.request.user
-        shift_facility = shift.facility
-        shift_organization = shift.facility.organization
-
-        if (is_manager(current_user, shift_organization, shift_facility)
-            or current_user.is_superuser):
+        facility = shift.facility
+        
+        if (self.can_delete_shift(current_user, facility)):
             return shift
         else:
             raise Http404
 
     def get_success_url(self):
         return reverse("shift_management")
+        
+    def can_delete_shift(self, user, facility):
+        """
+        Check if user has the general deletion rights and belongs
+        to the given facility.
+        """
+        if (not user.has_perm('scheduler.delete_shift')):
+            return False
+        user_facilities = get_cached_memberships(user)[1]
+        for facility_id in user_facilities:
```

Comment:
ah, I misread it. It was dubious since it uses `for` loop for the membership test instead of `in` – unlike your previous one. so I wondered why. Anyways, never mind!